daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs

```diff
@@ -123,13 +123,16 @@ getDalf :: NormalizedFilePath -> Action (Maybe LF.Package)
 getDalf file = eitherToMaybe <$>
     (runExceptT $ useE GeneratePackage file)
 
+getDalfModule :: NormalizedFilePath -> Action (Maybe LF.Module)
+getDalfModule file = eitherToMaybe <$> (runExceptT $ useE GenerateDalf file)
```

Comment:
Could you use `use` instead directly - that already returns a Maybe, so you don't have to wrap it in an either, run it in an either monad, then convert to a maybe.

Reply:
Good point, I changed it. We should go through some of the others, e.g., `getDalf` above and change them. The current logic doesn’t make much sense.

Reply:
Indeed. `useE` is an odd beast.